gridData/core.py

```diff
@@ -552,18 +550,12 @@ def _export_dx(self, filename, type=None, typequote='"', **kwargs):
         )
         dx = OpenDX.field('density', components=components, comments=comments)
         dx.write(filename)
-
-    def _export_dxgz(self, filename, type=None, typequote='"', **kwargs):
-        self._export_dx(filename, type, typequote, **kwargs)
-        root, ext = os.path.splitext(filename)
-        filename = root + '.dx'
-
-        with open(filename, 'rb') as in_file:
-            gz_filename = filename + '.gz'
-            with gzip.open(gz_filename, 'wb') as out_file:
-                for l in in_file:
-                    out_file.write(l)
-        os.remove(filename)
+        if ext == '.gz':
```

Comment:
I moved the logic to dx.write but given how it was originally implemented it was a bit tricky. I needed to add few isinstance to check if I needed to write string or bytes depending on the stream

Reply:
Also needed to implement a simple version of your 'openany' to deal with it:

https://github.com/eloyfelix/GridDataFormats/blob/DXGZ_input_format/gridData/OpenDX.py#L486

Reply:
I see. I don't particularly like having to repeat the same code everywhere. But that's because of the poor structure of the original code (i.e., my fault ;-)  where writing is actually done in each element of the DXfield. Can you think of a better code structure where we only need to do the encoding etc once? E.g., instead of having each element write, return a text representation that is then concatenated at the top level.

Reply:
added write_line function: https://github.com/eloyfelix/GridDataFormats/blob/DXGZ_input_format/gridData/OpenDX.py#L193

It's probably not optimal but it's the best that I could do without doing major codebase changes (which I'm not sure I'm ready to do)

Reply:
Your approach is fine, I just added a few minor change requests to clean up the code. Importantly, this is really cleaning up my mess – I would be very grateful if you could do it as part of this PR!